src/hotspot/share/runtime/thread.cpp

```diff
@@ -574,6 +574,10 @@ bool Thread::set_as_starting_thread(JavaThread* jt) {
 // short-duration critical sections where we're concerned
 // about native mutex_t or HotSpot Mutex:: latency.
 
+bool Thread::TrySpinAcquire(volatile int * adr) {
+  return AtomicAccess::cmpxchg(adr, 0, 1) == 0;
```

Comment:
How is this a try-spin-acquire operation ??? I don't think we need this, we can just inline the `cmpxchg` where needed.

Reply:
Right, not sure why I used that class. I replaced it with the atomic operations.